### ISSUE TITLE
show marked days with events

### ISSUE THREAD

**Issue by rafaxplayer** [twinssbc/Ionic2-Calendar#494]
Hello partner, I have problems in my app to show marked days with events.
My app is training in the gym and I only use the event system to know what day training has been done, so the starttime, endtime ranges do not make sense in the app.
I use this code to load the events, I think the problem is due to the format of dates or something like that:
ngOnInit () {
    this.subscriptionTraining = this.database.getTrainings (). subscribe ((trs: Training []) => {
      this.entrenos = trs;
      this.entrenos.forEach ((training: Training) => {
        let trDate = new Date (training.date);
        this.entrenosEvent.push ({
          title: "I train:" + training.date,
          startTime: Date.UTC (trDate.getUTCFullYear (), trDate.getUTCMonth (), trDate.getUTCDate ()),
          endTime: Date.UTC (trDate.getUTCFullYear (), trDate.getUTCMonth (), trDate.getUTCDate () + 1),
          allDay: true
        });
      });
      console.log (this.entrenosEvent);
      this.myCalendar.loadEvents ();
    });
    
  }
thanks for everything.

**Comment by twinssbc** [comment]
@rafaxplayer The code looks good to me. What's the issue? The event is not displayed or not in the expected date? Could you try push some dummy events to this.entrenosEvent instead of fetching events from database, see if it works?

**Comment by rafaxplayer** [comment]
The problem is that the days with calendar events are not marked.
One question I have is if the same date is established for the beginning as for the end, is it a valid event? Should it be marked on the calendar?

**Comment by twinssbc** [comment]
@rafaxplayer No, the start date and end date should not be the same if you are using allDay event.

**Comment by rafaxplayer** [comment]
So how could I do it so they show up? , my events are like this:
```
{
title: "I train: Tue Jan 28 2020"
startTime: 1580083200000
endTime: 1580083200000
allDay: true
}
```
There is no different end time because my app does not need anything more than the date the training took place.

My code to fill events is now like this:

```
this.database.getAllTrainings (). then ((trainings: Training []) => {

       trainings.forEach ((training: Training) => {

         let trDate = new Date (training.date);
         this.entrenosEvent.push ({
           title: "I train:" + training.date,
           startTime: Date.UTC (trDate.getUTCFullYear (), trDate.getUTCMonth (), trDate.getUTCDate ()),
           endTime: Date.UTC (trDate.getUTCFullYear (), trDate.getUTCMonth (), trDate.getUTCDate ()),
           allDay: false
         });
       });
       this.myCalendar.loadEvents ();
       console.log (this.entrenosEvent);
     });
```
Thanks spor respond so soon, greetings

**Comment by twinssbc** [comment]
@rafaxplayer If you set allDay to true, you can set endTime one day later than the startTime. If you set allDay to false, you also need to set the endTime later than startTime, they can't be the same. You could first try using some dummy event before converting events from your database.

**Comment by rafaxplayer** [comment]
Hello, I still can show the events in my calendar, my objects have this format:
{
title: "Training: 2020-01-30T23: 00: 00.000Z"
startTime: 1580342400000
endTime: 1580342401000
allDay: true
}
I added a second in endtime to eviatr errors but
and in console I have this error:
"ERROR Error: Uncaught (in promise): TypeError: event_1.startTime.getTime is not a function
TypeError: event_1.startTime.getTime is not a function at MonthViewComponent.push ../ node_modules / ionic2-calendar / monthview.js.MonthViewComponent.onDataLoaded "

any suggestions?

**Comment by twinssbc** [comment]
@rafaxplayer This is because startTime and endTime accepts a Date object, instead of number. You could set time like below 
```
{
title: "Training: 2020-01-30T23: 00: 00.000Z",
startTime: new Date(Date.UTC(2020,1,4)),
endTime: new Date(Date.UTC(2020,1,5)),
allDay: true
}
```

**Comment by rafaxplayer** [comment]
I already did it but I can't get the events marked on the calendar.
My objects are like this now:
```
{
title: "Entreno :2020-01-27"
startTime: Mon Jan 27 2020 01:00:00 GMT+0100 (hora estándar de Europa central) {}
endTime: Mon Jan 27 2020 01:00:01 GMT+0100 (hora estándar de Europa central)
__proto__: Object
allDay: true
}
```
and my code for loading them is this:

```
ngOnInit() {

    this.subscriptionTraining = this.database.getTrainings().subscribe((trs:Training[])=>{
      this.entrenos = trs;
      console.log('On Init Home Page Entrenos:',this.entrenos);
                 
    });

    this.database.getAllDates().then((dates:any[])=>{
      dates.forEach(( dat:any )=>{
        let trDate = new Date(dat.date);
         this.entrenosEvent.push({
          title: "Entreno :" + dat.date,
          startTime: new Date(Date.UTC(trDate.getUTCFullYear(),trDate.getUTCMonth(),trDate.getUTCDate(),0,0,0)),
          endTime: new Date(Date.UTC(trDate.getUTCFullYear(),trDate.getUTCMonth(),trDate.getUTCDate(),0,0,1)),
          allDay: true
        }); 
        
      }); 
      this.myCalendar.loadEvents();
      console.log(this.entrenosEvent);
    }); 
  }
```

I don't understand why it doesn't work ...

**Comment by twinssbc** [comment]
@rafaxplayer I think it's because you set the startTime and endTime to be the same day.
Try this
```
          endTime: new Date(Date.UTC(trDate.getUTCFullYear(),trDate.getUTCMonth(),trDate.getUTCDate()+1,0,0,0)),
```
